docs: add AI attribution policy (#848)
generative-computing/mellea: AGENTS.md

Reviewer: ```suggestion
- **Issue/workaround?** → Add to Section 9 (Common Issues) in this file
```
The section moved (you may want to replace with a proper link)

Author: fixed

Pre-commit runs: ruff, mypy, uv-lock, codespell

## 7. AI Attribution

> The project requires sign-offs — use `-s` when committing. Do not add a separate `Signed-off-by` line in the AI's own name. Add an `Assisted-by:` trailer to the commit footer by default:

Reviewer: this reads as a little confusing - it seems to tell an AI. to use -s when commiting and to not add a line, yet by definition it is what -s does. Maybe something like this

```suggestion
Commits require a Signed-off-by trailer from the human author (the human
runs `git commit -s`). AI agents must not add a Signed-off-by in the
tool's own name — instead, add an Assisted-by: trailer to the commit
footer by default:
```

My AI agent complained on this ambiguity.

Reviewer 2: This could be a little confusing too, it could be read as saying that the human will be the one to run the git commit command. Maybe something like this:

```
Commits require a Signed-off-by trailer from the human author, which is done 
by running `git commit -s`. Do not add an additional Signed-off-by ...
```

Author: I've made some changes based on this feedback if you'd like to take another look

Reviewer: Even tha text @psschwei since in many cases it's the agent creating the commit ... (I accept i omitted the role - because of this)

Reviewer: @ajbozarth I'm fine with your proposal -

Pre-commit runs: ruff, mypy, uv-lock, codespell

## 7. AI Attribution

> AI tools are welcome for development work in this project.

Reviewer 2: not-serious nit: given this is in AGENTS.md, this is redundant 😄

Author: nice catch, I'll fix that

Pre-commit runs: ruff, mypy, uv-lock, codespell

> For AI attribution trailers, see [§7 AI Attribution](#7-ai-attribution).

Author: ```suggestion
For AI attribution trailers, see [Section 7 (AI Attribution)](#7-ai-attribution).
```
